### PR TITLE
feat: [CGC-13] Add z-score values to line endpoints

### PR DESCRIPTION
**Pull request by edvinstava**

### CODE REVIEW

**Review by petterav — APPROVED** [review]

File: src/Components/GrowthChart/GrowthChartBuilder/GrowthChartBuilder.tsx [hunk 1]
```diff
@@ -24,7 +27,23 @@ export const GrowthChartBuilder = ({
...
+            const meta = chart.getDatasetMeta(index);
+            const [lastElement] = meta.data.slice(-1);
+            const { x, y } = lastElement.getProps(['x', 'y']);
+            ctx.fillStyle = typeof dataset.borderColor === 'string' ? dataset.borderColor : 'black';
+            ctx.fillText(dataset.label, x + 10, y);
+        });
     };
 
+    useEffect(() => {
+        Chart.register({ id: 'custom_draw', afterDraw });
```
Comment (henrikmv) [inline src/Components/GrowthChart/GrowthChartBuilder/GrowthChartBuilder.tsx:45]: Can you do this without Chart.register() please?

**Review by henrikmv — COMMENTED** [review]
A decent start to a difficult issue, but we need to make it closer to what is best practice

File: src/Components/GrowthChart/GrowthChartBuilder/GrowthChartBuilder.tsx [hunk 2]
```diff
@@ -14,16 +15,19 @@ export const GrowthChartBuilder = ({
             data: dataSetValues.map((entry) => entry[key]),
             borderWidth: 0.9,
             borderColor: chartLineColorPicker(key),
+            label: key,
         })),
     };
 
-    const options = {
+    const options: any = {
```
Comment (henrikmv) [inline src/Components/GrowthChart/GrowthChartBuilder/GrowthChartBuilder.tsx:22]: Remove any type for options

File: src/Components/GrowthChart/GrowthChartBuilder/GrowthChartBuilder.tsx [hunk 3]
```diff
@@ -14,16 +15,19 @@ export const GrowthChartBuilder = ({
...
         elements: { point: { radius: 0, hoverRadius: 0 } },
         plugins: { legend: { display: false } },
         scales: {
             x: { title: { display: true, text: i18n.t(`age (${dataSetMetadata.unit})`) } },
             y: { title: { display: true, text: dataSetMetadata.yaxis } },
         },
+        layout: { padding: { right: 75 } },
+        animation: { onProgress: annotateLineEnd },
```
Comment (henrikmv) [inline src/Components/GrowthChart/GrowthChartBuilder/GrowthChartBuilder.tsx:30]: After removing the any type for options, you will get an type error in option for the animation parameter. You can solve this by using ES6+ syntax, and defining type for `animation`
```suggestion
    animation: {
    onProgress(animation: AnimationForChart) {
        annotateLineEnd(animation);
    }
}
```

Comment (henrikmv) [reply src/Components/GrowthChart/GrowthChartBuilder/GrowthChartBuilder.tsx:30]: Maybe add this to `chartDataTypes.ts``
```suggestion
import { Animation, Chart } from 'chart.js';

interface AnimationWithChart extends Animation {
    chart?: Chart;
}
```

**Review by henrikmv — APPROVED** [review]
Well done @edvinstava! This looks good. Can you remove the any type for **options** before you squash merge.

**Pull request merged**